src/evaluation/mod.rs

```diff
@@ -0,0 +1,79 @@
+use std::collections::HashMap;
+
+#[derive(Clone)]
+pub struct EvaluationContext {
+    pub targetting_key: String,
+    pub attributes: HashMap<String, String>,
+}
+
+pub struct EvaluationOptions {}
+
+pub type FlattenedContext = HashMap<String, String>;
+
+pub fn flatten_context(context: EvaluationContext) -> FlattenedContext {
+    let mut flattened_context = HashMap::new();
+    flattened_context.insert(
```

Comment:
By merging these, we'll end up with a key and value which are equal and set to the targeting key. I haven't checked the spec, but this is different than how we do it in java. This implementation would mean that you should be careful about collisions between context keys and targeting keys.

Reply:
You're right, I actually just half copiloted this whilst I was transposing it from golang, What's the desired outcome here? We put the struct into a K/V map?

Reply:
Okay, updated LMK if that looks better - I took the golang impl